### ISSUE TITLE
MURA 7 Installation Issues - "Element default is undefined"

### ISSUE THREAD

**Issue by snovak70** [blueriver/MuraCMS#2398]
Hello! I hope someone can help me. I have had some major issues installing Mura 7 on our server. I figured out one issue but I am stumped on this one and I do cannot find any reference to it on the public support.

So I got Mura to install (at least it says it is installed) and when I click on the link to log in or try to go to the main login page (http://[our mura url]/admin/) I continually get an error:

500 Error
Element default is undefined in a CFML structure referenced as part of an expression. 

Has anyone else experienced this or know of a way to resolve this? Any help is appreciated!

**Comment by mattlevine** [comment]
If you're using Adobe CF could you enable robust degubbing info and post the stacktrace?

**Comment by snovak70** [comment]
Here you go Matt:

500 Error
Element default is undefined in a CFML structure referenced as part of an expression. 
Type: Expression 
 coldfusion.runtime.UndefinedElementException: Element default is undefined in a CFML structure referenced as part of an expression.
	at coldfusion.runtime.CfJspPage.ArrayGetAt(CfJspPage.java:972)
	at coldfusion.runtime.CfJspPage._arrayGetAt(CfJspPage.java:983)
	at coldfusion.runtime.CfJspPage._arrayGetAt(CfJspPage.java:978)
	at cfsettingsManager2ecfc1119076302$funcGETSITE.runFunction(E:\inetpub\MuraCMS\requirements\mura\settings\settingsManager.cfc:451)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:231)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:643)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:432)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:402)
	at coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:2483)
	at cfresourceBundleFactory2ecfc523582771$funcRESETSESSIONLOCALE.runFunction(E:\inetpub\MuraCMS\requirements\mura\resourceBundle\resourceBundleFactory.cfc:855)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:231)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:643)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:432)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:402)
	at coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:2483)
	at cfApplication2ecfc1872457087$funcSETUPREQUEST._factor3(E:\inetpub\MuraCMS\admin\Application.cfc:354)
	at cfApplication2ecfc1872457087$funcSETUPREQUEST.runFunction(E:\inetpub\MuraCMS\admin\Application.cfc:351)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:231)
	at coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:2845)
	at cfframework2ecfc2017272302$funcSETUPREQUESTWRAPPER.runFunction(E:\inetpub\MuraCMS\admin\framework.cfc:1221)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:231)
	at coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:2845)
	at cfframework2ecfc2017272302$funcONREQUESTSTART.runFunction(E:\inetpub\MuraCMS\admin\framework.cfc:641)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:533)
	at coldfusion.runtime.CfJspPage._invokeUDF(CfJspPage.java:2810)
	at coldfusion.runtime.SuperScope.invoke(SuperScope.java:18)
	at coldfusion.runtime.CfJspPage._invoke(CfJspPage.java:2424)
	at cfApplication2ecfc1872457087$funcONREQUESTSTART.runFunction(E:\inetpub\MuraCMS\admin\Application.cfc:255)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:487)
	at coldfusion.filter.SilentFilter.invoke(SilentFilter.java:47)
	at coldfusion.runtime.UDFMethod$ArgumentCollectionFilter.invoke(UDFMethod.java:383)
	at coldfusion.filter.FunctionAccessFilter.invoke(FunctionAccessFilter.java:95)
	at coldfusion.runtime.UDFMethod.runFilterChain(UDFMethod.java:334)
	at coldfusion.runtime.UDFMethod.invoke(UDFMethod.java:231)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:643)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:432)
	at coldfusion.runtime.TemplateProxy.invoke(TemplateProxy.java:402)
	at coldfusion.runtime.AppEventInvoker.invoke(AppEventInvoker.java:108)
	at coldfusion.runtime.AppEventInvoker.onRequestStart(AppEventInvoker.java:279)
	at coldfusion.filter.ApplicationFilter.invoke(ApplicationFilter.java:455)
	at coldfusion.filter.RequestMonitorFilter.invoke(RequestMonitorFilter.java:42)
	at coldfusion.filter.MonitoringFilter.invoke(MonitoringFilter.java:40)
	at coldfusion.filter.PathFilter.invoke(PathFilter.java:142)
	at coldfusion.filter.ExceptionFilter.invoke(ExceptionFilter.java:94)
	at coldfusion.filter.ClientScopePersistenceFilter.invoke(ClientScopePersistenceFilter.java:28)
	at coldfusion.filter.BrowserFilter.invoke(BrowserFilter.java:38)
	at coldfusion.filter.NoCacheFilter.invoke(NoCacheFilter.java:58)
	at coldfusion.filter.GlobalsFilter.invoke(GlobalsFilter.java:38)
	at coldfusion.filter.DatasourceFilter.invoke(DatasourceFilter.java:22)
	at coldfusion.filter.CachingFilter.invoke(CachingFilter.java:62)
	at coldfusion.CfmServlet.service(CfmServlet.java:219)
	at coldfusion.bootstrap.BootstrapServlet.service(BootstrapServlet.java:89)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:303)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at coldfusion.monitor.event.MonitoringServletFilter.doFilter(MonitoringServletFilter.java:42)
	at coldfusion.bootstrap.BootstrapFilter.doFilter(BootstrapFilter.java:46)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)
	at sun.reflect.GeneratedMethodAccessor582.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.intergral.fusionreactor.j2ee.filterchain.WrappedFilterChain.doFilter(WrappedFilterChain.java:97)
	at com.intergral.fusionreactor.j2ee.filter.FusionReactorRequestHandler.doNext(FusionReactorRequestHandler.java:472)
	at com.intergral.fusionreactor.j2ee.filter.FusionReactorRequestHandler.doHttpServletRequest(FusionReactorRequestHandler.java:312)
	at com.intergral.fusionreactor.j2ee.filter.FusionReactorRequestHandler.doFusionRequest(FusionReactorRequestHandler.java:192)
	at com.intergral.fusionreactor.j2ee.filter.FusionReactorRequestHandler.handle(FusionReactorRequestHandler.java:507)
	at com.intergral.fusionreactor.j2ee.filter.FusionReactorCoreFilter.doFilter(FusionReactorCoreFilter.java:36)
	at sun.reflect.GeneratedMethodAccessor580.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.intergral.fusionreactor.j2ee.filterchain.WrappedFilterChain.doFilter(WrappedFilterChain.java:79)
	at sun.reflect.GeneratedMethodAccessor566.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.intergral.fusionreactor.agent.filter.FusionReactorStaticFilter.doFilter(FusionReactorStaticFilter.java:53)
	at com.intergral.fusionreactor.agent.pointcuts.NewFilterChainPointCut$1.invoke(NewFilterChainPointCut.java:41)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:220)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:122)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:505)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:169)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:116)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:450)
	at org.apache.coyote.ajp.AjpProcessor.process(AjpProcessor.java:197)
	at org.apache.coyote.AbstractProtocol$AbstractConnectionHandler.process(AbstractProtocol.java:625)
	at org.apache.tomcat.util.net.JIoEndpoint$SocketProcessor.run(JIoEndpoint.java:316)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Unknown Source)
 

Column: 0
ID: ??
Line: 451
Raw Trace: at cfsettingsManager2ecfc1119076302$funcGETSITE.runFunction(E:\inetpub\MuraCMS\requirements\mura\settings\settingsManager.cfc:451)
Template: E:\inetpub\MuraCMS\requirements\mura\settings\settingsManager.cfc
Type: CFML

Column: 0
ID: CF_TEMPLATEPROXY
Line: 855
Raw Trace: at cfresourceBundleFactory2ecfc523582771$funcRESETSESSIONLOCALE.runFunction(E:\inetpub\MuraCMS\requirements\mura\resourceBundle\resourceBundleFactory.cfc:855)
Template: E:\inetpub\MuraCMS\requirements\mura\resourceBundle\resourceBundleFactory.cfc
Type: CFML

Column: 0
ID: CF_TEMPLATEPROXY
Line: 354
Raw Trace: at cfApplication2ecfc1872457087$funcSETUPREQUEST._factor3(E:\inetpub\MuraCMS\admin\Application.cfc:354)
Template: E:\inetpub\MuraCMS\admin\Application.cfc
Type: CFML

Column: 0
ID: CF_APPLICATION
Line: 351
Raw Trace: at cfApplication2ecfc1872457087$funcSETUPREQUEST.runFunction(E:\inetpub\MuraCMS\admin\Application.cfc:351)
Template: E:\inetpub\MuraCMS\admin\Application.cfc
Type: CFML

Column: 0
ID: CF_UDFMETHOD
Line: 1221
Raw Trace: at cfframework2ecfc2017272302$funcSETUPREQUESTWRAPPER.runFunction(E:\inetpub\MuraCMS\admin\framework.cfc:1221)
Template: E:\inetpub\MuraCMS\admin\framework.cfc
Type: CFML

Column: 0
ID: CF_UDFMETHOD
Line: 641
Raw Trace: at cfframework2ecfc2017272302$funcONREQUESTSTART.runFunction(E:\inetpub\MuraCMS\admin\framework.cfc:641)
Template: E:\inetpub\MuraCMS\admin\framework.cfc
Type: CFML

Column: 0
ID: CF_SUPERSCOPE
Line: 255
Raw Trace: at cfApplication2ecfc1872457087$funcONREQUESTSTART.runFunction(E:\inetpub\MuraCMS\admin\Application.cfc:255)
Template: E:\inetpub\MuraCMS\admin\Application.cfc
Type: CFML

**Comment by snovak70** [comment]
Hey there Matt, By chance did the the stack trace help at all? Anyone have any suggestions?

**Comment by mattlevine** [comment]
Unfortunately, it didn't give me much info.  If you look into the data does tsettings have a record where siteid equals 'default'?  Or seems like the application hasn't started?

-Matt

**Comment by snovak70** [comment]
There is no data in the table of tsettings.  I'm following the installation instructions. Any ideas?

**Comment by mattlevine** [comment]
The data in that table should have been inserted during the install process. So if there's no data in your tsettings table then I would start the install process over again from scratch with a new db.

**Comment by snovak70** [comment]
I have tried at least 2 dozen times with a fresh install and new empty database. I always seem to get this error on my first attempt of installation (see below). Then go through the process again as instructed on the MURA installation page and I get to "Installation Successful" then I get the error at the top of this thread.

Error:
Error during database creation (2). Please ensure you have updated drivers or contact support:Error Executing Database Query. - [Macromedia][SQLServer JDBC Driver][SQLServer]Incorrect syntax near '<'.

I have followed info I have found online specifically [https://groups.google.com/forum/#!searchin/mura-cms-developers/Error$3A$20Error$20during$20database$20creation$20(2).$20Please$20ensure$20you$20have$20updated$20drivers$20or$20contact$20support$3AError$20Executing$20Database$20Query.$20-$20%5BMacromedia%5D%5BSQLServer$20JDBC$20Driver%5D%5BSQLServer%5DIncorrect$20syntax$20near$20$27%3C$27.%7Csort:relevance/mura-cms-developers/diOBcQ-MAAU/hiNhc46keB0J](url).

I have added .SQL to the allowed file extension for CFINCLUDE.
We are not using sandbox security on this server.
The CF service has proper permissions.
We are running CF11 Standard on Windows Server 2012 with MSSQL 2012.

I am getting frustrated on all the time I am wasting on just trying to evaluate MURA for our needs and I cant get past the installation process. Any other ideas?

**Comment by snovak70** [comment]
So I turned on Debugging Info and when I go to the initial load of the Installation Process here are the errors that come up in the debug code. There is NO page level error thrown by the application (which would be useful in figuring this out). "E:/inetpub/MuraCMS/" is the root folder that Mura was installed in.  

I opened up the code on applicationSettings.cfm and then also the ini file and it looks like it is looking for a variable in the ini file called "context" that is used for the template path. In the ini file there is no value for "context".

Does this info help?

Exceptions
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 137
	    Could not find the included template /config/mappings.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 145
	    Could not find the included template /plugins/mappings.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 240
	    Could not find the included template /config/cfapplication.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 305
	    Could not find the included template /plugins/cfapplication.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 57
	    Could not find the included template config/mappings.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 79
	    Could not find the included template plugins/mappings.cfm.
	    
14:12:37.037 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 101
	    Could not find the included template plugins/cfapplication.cfm.
	    
14:12:38.038 - Expression Exception - in E:/inetpub/MuraCMS/config/appcfc/onRequestStart_include.cfm : line 65
	    Element SETTINGSMANAGER is undefined in APPLICATION.

**Comment by mattlevine** [comment]
This the only error that really matters:

```Error:
Error during database creation (2). Please ensure you have updated drivers or contact support:Error Executing Database Query. - [Macromedia][SQLServer JDBC Driver][SQLServer]Incorrect syntax near '<'.
```

One that happens then the db it not properly populated, which in turn creates the second error.

I asked our team to try and see if we can recreate the issue in order to try an understand what is happening.  Can you look into your /config directory and see if there's an "errors" directory in it?  If there is please zip then an post them.

-Matt

**Comment by gowestweb** [comment]
@snovak70  I'm checking into it. Please provide the full error details if possible, so we can see which line specifically is throwing the SQL syntax error, thanks.

**Comment by snovak70** [comment]
So I just attempted a fresh install with a new database. When I got to the very first page of the installation process I get the following debug errors. I did not do anything but load the page. It appears there is a file mapping issue and this appears before the SQL error message: I also attached a PDF of that page. I also attached a PDF of the debug of the setup process after I go through the 3 steps and then I get the SQL query error.  The line error is in /setup/inc/_process.cfm: Line 65 and line 232

The code on line 63 to 71 (_process.cfm) looks like it should only run if no datasource is provided in the setup process; however I do provide a datasouce name (which I set up in the CF Admin and verified) in step 1 of the installation process so in theory line 65 should not even run. Of course this is only from my quick observation.

[Mura CMS - post installation attempt.pdf](https://github.com/blueriver/MuraCMS/files/433975/Mura.CMS.-.post.installation.attempt.pdf)

[Mura CMS - Setup.pdf](https://github.com/blueriver/MuraCMS/files/433970/Mura.CMS.-.Setup.pdf)


Exceptions
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 137
	    Could not find the included template /config/mappings.cfm.
	    
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 145
	    Could not find the included template /plugins/mappings.cfm.
	    
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 240
	    Could not find the included template /config/cfapplication.cfm.
	    
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/config/applicationSettings.cfm : line 305
	    Could not find the included template /plugins/cfapplication.cfm.
	    
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 57
	    Could not find the included template config/mappings.cfm.
	    
22:12:27.027 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 79
	    Could not find the included template plugins/mappings.cfm.
	    
22:12:28.028 - MissingInclude Exception - in E:/inetpub/MuraCMS/Application.cfc : line 101
	    Could not find the included template plugins/cfapplication.cfm.
	    
22:12:28.028 - Expression Exception - in E:/inetpub/MuraCMS/config/appcfc/onRequestStart_include.cfm : line 65
	    Element SETTINGSMANAGER is undefined in APPLICATION.
